### ISSUE TITLE
usage: Report how useful weak hash is

### ISSUE THREAD

**Issue by AudriusButkevicius** [syncthing/syncthing#3806]
^^^

**Comment by calmh** [comment]
I started adding code to the rwfolder to do this but got bored and happened to wipe it out. What I started doing, if someone wants to pick it up, was

- pass the folderstatref from model to the folder factory function directly (it's cleaner)
- added a new thing in rwfolder to track the number of bytes copied, pulled and shifted
- add a method to the folder statistics thing to increment the values for a folder, called from the same place where we set the last received file or something similar

There are concurrency concerns to doing it that way though. Can also add the info about the number of changed bytes into the dbupdatejob perhaps. Or probably any number of other approaches. I first started putting it into the sharedPullerState, where we already track the number of copied blocks and so on.
